src/java/org/apache/nutch/tools/FileDumper.java

```diff
@@ -172,6 +172,11 @@ public boolean accept(File file) {
         }
       });
 
+      if (partDirs == null) {
+        LOG.warn("Skipping Corrupt Segment: [" + segment.getAbsolutePath() + "]");
```

Comment:
Note that this is not good syntax for slf4j. Line could be made much more performant by using
```
LOG.warn("Skipping Corrupt Segment: [{}]", segment.getAbsolutePath());
```
I am +1 to merge this, I will make the change locally and commit.